sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -288,8 +288,9 @@ private static boolean initConfigurations(final @NotNull SentryOptions options)
 
     final String dsn = options.getDsn();
     if (dsn == null) {
-      throw new IllegalArgumentException("DSN is required. Use empty string to disable SDK.");
-    } else if (dsn.isEmpty()) {
+      throw new IllegalArgumentException(
```

Comment:
We should check the `enabled` flag first and not throw if disabled regardless of DSN.

Reply:
Yeah, makes sense, any preference between the following two variants?

```
    if(!options.isEnabled() || (dsn != null && dsn.isEmpty())) {
      close();
      return false;
    } else if (dsn == null) {
      throw new IllegalArgumentException(
        "DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.");
    }
``` 

```
    if (!options.isEnabled()) {
      close();
      return false;
    }

    if (dsn == null) {
      throw new IllegalArgumentException(
          "DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.");
    } else if (dsn.isEmpty()) {
      close();
      return false;
    }
```

Reply:
Top one seems a bit better as it doesn't duplicate the code.